pennylane/devices/default_qubit_legacy.py

```diff
@@ -1086,6 +1087,6 @@ def _get_diagonalizing_gates(self, circuit: qml.tape.QuantumTape) -> List[Operat
         meas_filtered = [
             m
             for m in circuit.measurements
-            if m.obs is None or not isinstance(m.obs, qml.Hamiltonian)
+            if m.obs is None or not isinstance(m.obs, (qml.Hamiltonian, qml.ops.LinearCombination))
```

Comment:
Like discussed in the meeting, we want to stay away from `isinstance` checks with `qml.Hamiltonian` after Mudit's PR
```suggestion
            if m.obs is None or not isinstance(m.obs, (qml.ops.Hamiltonian, qml.ops.LinearCombination))
```